src/domain/entities/schemas/lenient-page.schema.factory.ts

```diff
@@ -0,0 +1,21 @@
+import { TypeOf, z } from 'zod';
+import { Page } from '@/domain/entities/page.entity';
+
+/**
+ * Builds a lenient page schema that filters out invalid items from the results array.
+ */
+export function buildLenientPageSchema<T extends z.ZodTypeAny>(
+  itemSchema: T,
+): z.ZodType<Page<z.infer<T>>> {
+  return z.object({
+    count: z.number().nullable(),
```

Comment:
Nit: this includes any erroneous `results` items in the `count`. Do you think we should reduce it accordingly?

Reply:
Sure! Good point!

Reply:
I've addressed this in a [separate PR.](https://github.com/safe-global/safe-client-gateway/pull/1881)